cmd/jujud/agent/machine.go

```diff
@@ -264,22 +269,29 @@ func MachineAgentFactoryFn(
 func NewMachineAgent(
 	machineId string,
 	agentConfWriter AgentConfigWriter,
-	bufferedLogs logsender.LogRecordCh,
+	bufferedLogger *logsender.BufferedLogWriter,
 	runner worker.Runner,
 	loopDeviceManager looputil.LoopDeviceManager,
+	newIntrospectionSocketName func(names.Tag) string,
 	rootDir string,
 ) *MachineAgent {
-	return &MachineAgent{
+	a := &MachineAgent{
 		machineId:                   machineId,
 		AgentConfigWriter:           agentConfWriter,
 		configChangedVal:            voyeur.NewValue(true),
-		bufferedLogs:                bufferedLogs,
+		bufferedLogger:              bufferedLogger,
 		workersStarted:              make(chan struct{}),
 		runner:                      runner,
 		rootDir:                     rootDir,
 		initialUpgradeCheckComplete: gate.NewLock(),
 		loopDeviceManager:           loopDeviceManager,
+		newIntrospectionSocketName:  newIntrospectionSocketName,
+		prometheusRegistry:          newPrometheusRegistry(),
 	}
+	a.prometheusRegistry.MustRegister(
```

Comment:
In general, not a fan of using something that can panic in non-test code. Could we make NewMachineAgent return an error?